### PR TITLE
Use k8s Duration instead of string for build timeout

### PR DESCRIPTION
**Pull request by shashwathi**
Fixes #335 

## Proposed Changes

  * Follow up PR to update type from string to k8s Duration type

cc @tcnghia

### CODE REVIEW

File: pkg/webhook/build.go [hunk 1]
```diff
@@ -237,19 +237,13 @@ func validationError(reason, format string, fmtArgs ...interface{}) error {
...
-		if err != nil {
-			return validationError("InvalidTimeFormat", "invalid build timeout %q err %#v. Refer https://golang.org/pkg/time/#ParseDuration for time format documentation", timeout, err)
-		}
-		if timeoutDuration > maxTimeout {
-			return validationError("InvalidTimeout", "build timeout exceeded 24h")
-		}
+	if timeout.Duration > maxTimeout {
+		return validationError("InvalidTimeout", "build timeout exceeded 24h")
+	} else if timeout.Duration < 0 {
+		return validationError("InvalidTimeFormat", "invalid build timeout %q. Refer https://golang.org/pkg/time/#ParseDuration for time format documentation", timeout)
```
Comment (ImJasonH) [inline pkg/webhook/build.go]: This type should handle parse validation errors now, so this error message should just be "cannot define negative timeout" or something along those lines.

File: pkg/webhook/webhook_test.go [hunk 2]
```diff
@@ -210,18 +211,18 @@ func TestValidateBuild(t *testing.T) {
 		reason: "invalid build timeout",
 		build: &v1alpha1.Build{
 			Spec: v1alpha1.BuildSpec{
-				Timeout: "garbagetimeout",
+				Timeout: metav1.Duration{Duration: -48 * time.Hour},
```
Comment (ImJasonH) [inline pkg/webhook/webhook_test.go]: This test case should be "negative timeout" now

File: pkg/webhook/build.go [hunk 3]
```diff
@@ -237,19 +237,13 @@ func validationError(reason, format string, fmtArgs ...interface{}) error {
...
-		if err != nil {
-			return validationError("InvalidTimeFormat", "invalid build timeout %q err %#v. Refer https://golang.org/pkg/time/#ParseDuration for time format documentation", timeout, err)
-		}
-		if timeoutDuration > maxTimeout {
-			return validationError("InvalidTimeout", "build timeout exceeded 24h")
-		}
+	if timeout.Duration > maxTimeout {
+		return validationError("InvalidTimeout", "build timeout exceeded 24h")
+	} else if timeout.Duration < 0 {
+		return validationError("InvalidTimeFormat", "build timeout should be greater than 0")
```
Comment (ImJasonH) [inline pkg/webhook/build.go]: The reason should be `InvalidFormat` just like a timeout >24h, otherwise everything LGTM

**Comment by pivotal-nader-ziada** [comment]
good straight forward change 
/lgtm

**Pull request merged**